api/schema_system.go

```diff
@@ -117,3 +158,24 @@ func UnmarshalNamingRulePayloadFormat(payload string) (*regexp.Regexp, error) {
 	}
 	return format, nil
 }
+
+// UnmarshalTemplateRulePayload will unmarshal payload to TemplateRulePayload and extract all the template keys.
+// For example, "hard_code_{{table}}_{{column}}_end" will return
+// "hard_code_{{table}}_{{column}}_end", ["{{table}}", "{{column}}"]
+func UnmarshalTemplateRulePayload(ruleType SchemaReviewRuleType, payload string) (string, []string, error) {
+	var nr NamingRulePayload
```

Comment:
`UnmarshalTemplateRulePayload` actually used `NamingRulePayload` instead of `TemplateRulePayload`.

So, how about substituting `UnmarshalNamingRulePayloadAsTemplate` for `UnmarshalTemplateRulePayload` and `UnamrshalNamingRulePayloadAsRegular` for `UnmarshalNamingRulePayloadFormat`?

Reply:
Sounds good. I rename them to `UnamrshalNamingRulePayloadAsRegexp` and `UnmarshalNamingRulePayloadAsTemplate`